From 79e352eb7869b3a32303d80860d15235a7e44606 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Sat, 19 Jan 2019 01:45:49 +0200
Subject: [PATCH 8/9] LineageSettingsProvider: Fix migration of
 FORCE_SHOW_NAVBAR

* It was moved from Global to System but no migration was in place
  to handle upgrades from previous versions.

Change-Id: If76243b0224dbc0138cf3c5dbc8aaeefa69f06a4
---
 .../lineagesettings/LineageDatabaseHelper.java        | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
index d381edf..da14c5b 100644
--- a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
+++ b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
@@ -51,7 +51,7 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
     private static final boolean LOCAL_LOGV = false;
 
     private static final String DATABASE_NAME = "lineagesettings.db";
-    private static final int DATABASE_VERSION = 10;
+    private static final int DATABASE_VERSION = 11;
 
     private static final String DATABASE_NAME_OLD = "cmsettings.db";
 
@@ -362,6 +362,15 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
             }
             upgradeVersion = 10;
         }
+
+        if (upgradeVersion < 11) {
+            // Move force_show_navbar to system
+            moveSettingsToNewTable(db, LineageTableNames.TABLE_GLOBAL,
+                    LineageTableNames.TABLE_SYSTEM, new String[] {
+                    LineageSettings.System.FORCE_SHOW_NAVBAR
+            }, true);
+            upgradeVersion = 11;
+        }
         // *** Remember to update DATABASE_VERSION above!
     }
 
-- 
2.17.1

